### DISCUSSION TITLE
How do you handle magic link failures with the PKCE auth flow and a server side callback?

### DISCUSSION THREAD

**Discussion by fnimick** [supabase/supabase#15196] [Questions]
With the PKCE auth flow, the user is redirected to a server endpoint which exchanges the provided token for a session.

When a magic link expires, it appears that the redirect has the error set in a url hash, which isn't sent to the server endpoint at all. There's no chance (I think) for the client to intercept this and render an error page. How can we detect an error in the server endpoint and show a message to the client accordingly?

For example - the magic link returns a 303 to `
http://localhost:5173/auth/callback#error=unauthorized_client&error_code=401&error_description=Email+link+is+invalid+or+has+expired`.

**Comment by fnimick** [comment]
Found it with help from a dev - it's waiting on PR https://github.com/supabase/gotrue/pull/1129

**Comment by J0** [comment]
Going to close since this has been merged. Thanks for flagging!
